### PR TITLE
PWN-6775 - Add Lokalise integration

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6775

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/lokalise/LokaliseManager.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import android.content.Context
+import com.lokalise.sdk.BuildConfig
+import com.lokalise.sdk.Lokalise
+import com.lokalise.sdk.LokaliseCallback
+import com.lokalise.sdk.LokaliseUpdateError
+import timber.log.Timber
+
+private const val TAG = "LokaliseManager"
+
+object LokaliseManager {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/lokalise/LokaliseManager.kt:12]: LokaliseService / LokaliseFacade

File: app/src/main/java/org/p2p/wallet/lokalise/LokaliseManager.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    fun setup(context: Context, lokaliseToken: String, projectId: String) {
+        with(Lokalise) {
+            init(context, lokaliseToken, projectId)
+            isPreRelease = BuildConfig.DEBUG
+            logsEnabled = BuildConfig.DEBUG
+            if (BuildConfig.DEBUG) {
+                addCallback(object : LokaliseCallback {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/lokalise/LokaliseManager.kt:19]: extract object to property / method

File: app/src/main/java/org/p2p/wallet/lokalise/LokaliseManager.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                    override fun onUpdateNotNeeded() {
+                        Timber.tag(TAG).i("No update needed")
+                    }
+
+                    override fun onUpdateFailed(error: LokaliseUpdateError) {
+                        Timber.tag(TAG).i("Error on fetching locale $error")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/lokalise/LokaliseManager.kt:25]: if LokaliseUpdateError is Throwable, use Timber.tag(TAG).i(error,"Error on fetching locale $error")

File: app/src/main/java/org/p2p/wallet/root/RootActivity.kt [hunk 4]
```diff
@@ -69,6 +70,10 @@ class RootActivity :
         handleDeeplink(intent)
     }
 
+    override fun attachBaseContext(newBase: Context) {
+        super.attachBaseContext(LokaliseContextWrapper.wrap(newBase))
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/root/RootActivity.kt:74]: To LokaliseFacade too

File: scripts/config.gradle [hunk 5]
```diff
@@ -18,6 +18,8 @@ android {
         buildConfigField("String", "intercomApiKey", "\"${apiKeysProperties["intercomApiKey"]}\"")
         buildConfigField("String", "intercomAppId", "\"${apiKeysProperties["intercomAppId"]}\"")
         buildConfigField("String", "appsFlyerKey", "\"${apiKeysProperties["appsFlyerKey"]}\"")
+        buildConfigField("String", "lokaliseKey", "\"${apiKeysProperties["lokaliseKey"]}\"")
+        buildConfigField("String", "lokaliseAppId", "\"${apiKeysProperties["lokaliseAppId"]}\"")
     }
```
Comment (gslevinkov) [inline scripts/config.gradle:23]: any gradle task to fetch new lokalise files from remote?

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: scripts/config.gradle (hunk 5 above)
Comment (Davran331122) [reply scripts/config.gradle:23]: no need

**Review by gslevinkov — APPROVED** [review]

**Review by NikolaiEmelianov — APPROVED** [review]

**Pull request merged**